ovs/lib/tests/vdisk_tests/test_events.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2017 iNuron NV
+#
+# This file is part of Open vStorage Open Source Edition (OSE),
+# as available from
+#
+#      http://www.openvstorage.org and
+#      http://www.openvstorage.com.
+#
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+# as published by the Free Software Foundation, in version 3 as it comes
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+
+"""
+Test module for vDisk events functionality
+"""
+import unittest
+from ovs.extensions.generic.configuration import Configuration
+from ovs.extensions.storage.persistentfactory import PersistentFactory
+from ovs.extensions.storage.volatilefactory import VolatileFactory
+from ovs.extensions.storageserver.storagedriver import MDSMetaDataBackendConfig, MDSNodeConfig
+from ovs.extensions.storageserver.tests.mockups import StorageRouterClient
+from ovs.lib.tests.helpers import Helper
+from ovs.lib.vdisk import VDiskController
+
+
+class VDiskEventsTest(unittest.TestCase):
+    """
+    This test class will validate various vDisk events functionality
+    """
+    @classmethod
+    def setUpClass(cls):
+        """
+        Sets up the unittest, mocking a certain set of 3rd party libraries and extensions.
+        This makes sure the unittests can be executed without those libraries installed
+        """
+        cls.persistent = PersistentFactory.get_client()
+        cls.persistent.clean()
+        cls.volatile = VolatileFactory.get_client()
+        cls.volatile.clean()
+        StorageRouterClient.clean()
+
+        Configuration.set('/ovs/framework/storagedriver|mds_tlogs', 100)
+        Configuration.set('/ovs/framework/storagedriver|mds_maxload', 75)
+        Configuration.set('/ovs/framework/storagedriver|mds_safety', 2)
+
+    def setUp(self):
+        """
+        (Re)Sets the stores on every test
+        """
+        # Cleaning storage
+        self.volatile.clean()
+        self.persistent.clean()
+        StorageRouterClient.clean()
+
+    def tearDown(self):
+        """
+        Clean up the unittest
+        """
+        # Cleaning storage
+        self.volatile.clean()
+        self.persistent.clean()
```

Comment:
Teardown should also call `StorageRouterClient.clean()`
or a TearDownClass should do so